packages/react/src/components/sideNavigation/mainLevel/MainLevel.tsx

```diff
@@ -159,10 +160,13 @@ export const MainLevel = ({
   const [isOpen, setIsOpen] = useState<boolean>(defaultOpenMainLevels.includes(index as number));
   const [isActiveParent, setIsActiveParent] = useState<boolean>(false);
 
-  const subLevels = React.Children.map(children, (child) => {
+  const childElements = getChildrenAsArray(children);
+
+  const subLevels = childElements.map((child, childIndex) => {
     if (isValidElement(child) && (child.type as FCWithName).componentName === 'SubLevel') {
       return cloneElement(child, {
         mainLevelIndex: index,
```

Comment:
This one also gives me a TS error. If these don't appear to you, then it's fine.

"No overload matches this call.
  The last overload gave the following error.
    Argument of type '{ mainLevelIndex: number; key: number; }' is not assignable to parameter of type 'Partial<unknown> & Attributes'.
      Object literal may only specify known properties, and 'mainLevelIndex' does not exist in type 'Partial<unknown> & Attributes'.ts(2769)"

Reply:
Interestingly it does not do that for me 🤔

Reply:
Ok, it's just something on my part then.